migrations/drop-audit-doc-index.js

```diff
@@ -0,0 +1,115 @@
+var db = require('../db'),
+    async = require('async');
+
+var DDOC_ID = '_design/medic';
+var BATCH_SIZE = 100;
+
+var dropView = function(callback) {
+  db.audit.get(DDOC_ID, function(err, ddoc) {
+    if (err) {
+      return callback(err);
+    }
+    delete ddoc.views;
+    db.audit.insert(ddoc, callback);
+  });
+};
+
+var needsUpdate = function(row) {
+  return row.doc.type === 'audit_record' && // is an audit doc, and
+         row.doc.record_id;                 // has old property
+};
+
+var createNewDocs = function(oldDocs, callback) {
+  var newDocs = oldDocs.map(function(doc) {
+    return {
+      _id: doc.record_id + '-audit',
+      type: 'audit_record',
+      history: doc.history
+    };
+  });
+  db.audit.bulk({ docs: newDocs }, callback);
+};
+
+var deleteOldDocs = function(oldDocs, callback) {
+  oldDocs.forEach(function(doc) {
+    doc._deleted = true;
+  });
+  db.audit.bulk({ docs: oldDocs }, callback);
+};
+
+var changeDocIdsBatch = function(skip, callback) {
+  var options = {
+    include_docs: true,
+    limit: BATCH_SIZE,
+    skip: skip
+  };
+  db.audit.list(options, function(err, result) {
+    if (err) {
+      return callback(err);
+    }
+    if (!result.rows || !result.rows.length) {
+      // we've reached the end of the database!
+      return callback(null, null, false);
+    }
+    console.log('        Processed ' + skip + ' docs of ' + result.total_rows + ' total');
+    var oldDocs = result.rows.filter(needsUpdate).map(function(row) {
+      return row.doc;
+    });
+    if (!oldDocs.length) {
+      // no old docs in this batch
+      return callback(null, skip + BATCH_SIZE, true);
+    }
+    createNewDocs(oldDocs, function(err) {
+      if (err) {
+        return callback(err);
+      }
+      deleteOldDocs(oldDocs, function(err) {
+        if (err) {
+          return callback(err);
+        }
+        // The newSkip is based on the old skip, plus the number of
+        // unchanged files. Changed files can't be included because
+        // they have just been deleted so will not be included in the
+        // next query result. This means some records will be processed
+        // more than once but gaurantees every record will be processed
+        // at least once.
+        var newSkip = skip + BATCH_SIZE - oldDocs.length;
```

Comment:
If this encounters a run of 100 records, all of which are missing a `record_id` or have a type other than `audit_record`, it looks like this would loop infinitely. If I'm reading this right, I'd probably suggest that we enforce some kind of forward progress (e.g. `skip + max(BATCH_SIZE - oldDocs.length, 1)`).